File: TwitterRedditProtoype/TwitterRedditProtoype/ContentPiece.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lurk
{
    abstract class ContentPiece
    {
        protected DateTime timeStamp;

        public abstract void Display();

        public static int CompareByDate(ContentPiece piece1, ContentPiece piece2)
        {
            return piece1.timeStamp.CompareTo(piece2.timeStamp);
        }
    }
}

File: TwitterReddit/app/src/main/java/com/example/twitterreddit/ContentPiece.java
package com.example.twitterreddit;

import java.time.LocalDateTime;

abstract class ContentPiece implements Comparable<ContentPiece>
{
    public LocalDateTime timeStamp;

    public abstract void Display();

    public static int compareTo(ContentPiece piece1, ContentPiece piece2)
    {
        return piece1.timeStamp.compareTo(piece2.timeStamp);
    }
}